design_drafts/joint_interfaces_and_transmissions.md

```diff
@@ -35,49 +35,57 @@ and ties in with the use of the [flexible joint states message](https://github.c
+        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
+        <!-- output state (x_dot_dot) from the joint, e.g. acceleration -->
+        <hardwareInterface>hardware_interface/AccelerationJointInterface</hardwareInterface>
+      </sensor>
+    </joint>
+    <joint name="joint2">
+      <actuator>
+        <hardwareInterface>hardware_interface/EffortJointCommandInterface</hardwareInterface>
+      </actuator>
+      <sensor>
+        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
+        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
+        <hardwareInterface>hardware_interface/AccelerationJointInterface</hardwareInterface>
+      </actuator>
+    </joint>
+    <joint>
+      <actuator>
+        <!-- control input (u) for the joint, i.e. velocity value -->
+        <hardwareInterface>hardware_interface/VelocityJointCommandInterface</hardwareInterface>
+      </actuator>
+      <sensor>
+        <!-- custom output state (x) for the joint, i.e. some custom value such as current      
+        <hardwareInterface name="current_joint_interface">hardware_interface/CustomValueInterface</hardwareInterface>
+      </sensor>
     </joint>
-    <actuator name="actuator1">
-      <mechanicalReduction>50</mechanicalReduction>
-    </actuator>
 </transmission>
 ```
-The content of a `hardwareInterface`  XML attribute should match up with strings provided by `ros2_control` in the shape of familiar names:
+
+The content of a `hardwareInterface` XML tag should match up with strings provided by `ros2_control` in the shape of familiar names:
 `hardware_interface::PositionJointInterface, hardware_interface::VelocityJointInterface, hardware_interface::EffortJointInterface`.
-By relying on strings internally in the framework but using them mostly through shared names, a standard nomenclature is kept all the while allowing custom extensions in an easy and transparent way.
-Note the in the example above, since all names are merely strings, `awesome_interface/FooJointInterface` - a custom interface - can be used if it is provided by the `RobotHardware`.
+By relying on strings internally in the framework but using them mostly through shared names, a standard nomenclature is kept all the while allowing custom extensions in an easy and transparent way. 
 
 ### RobotHardware
 
-```
-void RobotHardware::declareInterface(string name);
-```
-or
-```
-void RobotHardware::registerInterface(string interface_name, string joint_name);
-```
-
-a possible automated way adding these to the `RobotHardware` is to declare them when 
-```
-template<class... interfaces>
-class RobotHardware{};
-...
-MyRobotHW : public RobotHardware<hardware_interface::PositionJointInterface, hardware_interface::VelocityJointInterface>
-```
+Based on the URDF snippet above, the instance of a robot hardware can be generated as described in the [hardware interface design doc](hardware_interface.md).
```

Comment:
This is largely misleading. The role of transmission parsing was never to set up an instance of robothw but to check whether a required transmission's requirements are met by the robothw which in this case serves as a HAL (hardware abstraction layer) and if it's green-lit, instantiate transmissions by "wiring" all relevant joint and actuator handles up.

However we provide a ton of tools for it and even reference (boilerplate) implementations for it, whoever implements `ros2_control` for their robot still holds the responsibility of setting up this HAL as that's territory we can't reasonably guess the details of.

I agree though that the content in this section is a bit of a braindump of mine.